image-hd.c

```diff
@@ -636,7 +636,7 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 			}
 		}
 		if (part->autoresize) {
-			long long partsize = image->size - now;
+			long long partsize = image->size - part->offset;
```

Comment:
Eh, it's not (just) a matter of whether now is a multiple of the alignment or not. If part->offset has been set explicitly (granted, that's a bit unlikely, since one must also have set the image size explicitly, so there's not much autoresize left to do...), the size is computed to be much too big. But yes, it should probably come sooner in the series, with a reworded commit message to phrase it (primarily) as a preparation for separate part->align values.